src/core/shape_inference/include/slice_shape_inference_utils.hpp

```diff
@@ -134,7 +134,11 @@ inline int64_t get_sliced_value(const int64_t& dim, const int64_t& start, const
     constexpr int64_t inf_bound = -1;
 
     const auto& norm_dim = dim == inf_bound ? std::numeric_limits<int64_t>::max() : dim;
+#if INTPTR_MAX == INT64_MAX
     const auto is_norm_dim_max = ov::internal::is_max(norm_dim);
+#else
+    const auto is_norm_dim_max = ov::internal::is_max(size_t(norm_dim));
+#endif
```

Comment:
Can we just change it to:
```suggestion
    const auto& norm_dim = dim == inf_bound ? std::numeric_limits<size_t>::max() : static_cast<size_t>(dim);
    const auto is_norm_dim_max = ov::internal::is_max(norm_dim);
```

Reply:
It seems it doesn't work, because it will bring more other failures in 32bits build:

[  FAILED  ] 8 tests, listed below:
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_MAX_MIN_INT_dynamic_dimensions_neg_step
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_step
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_step_neg_ind
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_step_mix_ind
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_MAX_MIN_INT_64_dynamic_dimensions_neg_step
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_MAX_MIN_64_no_upper_bounds_neg_step
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_MAX_MIN_32_no_upper_bounds_neg_step
[  FAILED  ] type_prop.strided_slice_reverse_end_is_int64_min


```
[ RUN      ] type_prop.slice_v8_basic_const_inputs_MAX_MIN_INT_dynamic_dimensions_neg_step
/home/ov2022/workspace/wasm/openvino/src/core/tests/type_prop/slice.cpp:469: Failure
Expected equality of these values:
  op->get_output_partial_shape(0)
    Which is: { 8, 8..18, 5..15, ..9, ..9, ..4, 0, 30..40, 20, 20, ?, 21, 4294967296.. }
  expected_out_shape
    Which is: { 8, 8..18, 5..15, ..9, ..9, ..4, 0, 30..40, 20, 20, ..20, ..21, ? }
[ RUN      ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_step_mix_ind
/home/ov2022/workspace/wasm/openvino/src/core/tests/type_prop/slice.cpp:564: Failure
Expected equality of these values:
  op->get_output_partial_shape(0)
    Which is: { 6, 3, 6, ?, 4294967295.., 4294967296.. }
  expected_out_shape
    Which is: { ..6, ..3, ..6, ?, ?, ? }
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_step_mix_ind (1 ms)
```

Reply:
There are also some failures in 64bit build:

[  FAILED  ] 4 tests, listed below:
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_ind
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_step
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims_neg_step_neg_ind

```
[ RUN      ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims
/home/ov2022/workspace/capi/openvino/src/core/tests/type_prop/slice.cpp:488: Failure
Expected equality of these values:
  op->get_output_partial_shape(0)
    Which is: { 0, 0, 0, ? }
  expected_out_shape
    Which is: { ..6, ..15, ..5, ? }
[  FAILED  ] type_prop.slice_v8_basic_const_inputs_data_full_dynamic_dims (1 ms)
```

```
[ RUN      ] SliceStaticShapeInferenceTest.forward_step_all_data_in_const_map
/home/ov2022/workspace/capi/openvino/src/plugins/intel_cpu/tests/unit/shape_inference_test/slice_shape_inference_test.cpp:107: Failure
Expected equality of these values:
  output_shapes.front()
    Which is: { 10, 3, 0, 0, 18446744073709551615, 18446744073709551615, 3 }
  StaticShape({10, 3, 0, 4, max_d, max_d, 3})
    Which is: { 10, 3, 0, 4, 18446744073709551615, 18446744073709551615, 3 }
[  FAILED  ] SliceStaticShapeInferenceTest.forward_step_all_data_in_const_map (1 ms)
```